src/components/ButtonPanel.js

```diff
@@ -1,38 +1,125 @@
 import React from 'react';
 import Button from './Button';
 
-const ButtonPanel = () => (
-  <div className="Buttons">
-    <div className="group1">
-      <Button btname="AC" type="Button" color="gray" />
-      <Button btname="+/-" type="Button" color="gray" />
-      <Button btname="%" type="Button" color="gray" />
-      <Button btname="/" type="Button" />
-    </div>
-    <div className="group2">
-      <Button btname="7" type="Button" color="gray" />
-      <Button btname="8" type="Button" color="gray" />
-      <Button btname="9" type="Button" color="gray" />
-      <Button btname="X" type="Button" />
-    </div>
-    <div className="group3">
-      <Button btname="4" type="Button" color="gray" />
-      <Button btname="5" type="Button" color="gray" />
-      <Button btname="6" type="Button" color="gray" />
-      <Button btname="-" type="Button" />
-    </div>
-    <div className="group4">
-      <Button btname="3" type="Button" color="gray" />
-      <Button btname="2" type="Button" color="gray" />
-      <Button btname="1" type="Button" color="gray" />
-      <Button btname="+" type="Button" color="orange" />
-    </div>
-    <div className="group5">
-      <Button btname="0" type="Button" wide="1" color="gray" />
-      <Button btname="." type="Button" color="gray" />
-      <Button btname="=" type="Button" />
-    </div>
-  </div>
-);
+class ButtonPanel extends React.Component {
+  handle(btname) {
+    const prps = this.props;
+    const btName = btname;
+    /* eslint react/prop-types: 0 */
+    prps.onClick(btName);
+  }
+
+  render() {
+    return (
+      <div className="Buttons">
+        <div className="group1">
+          <Button
+            btname="AC"
+            type="Button"
+            color="gray"
+            onClick={() => this.handle('AC')}
```

Comment:
we want to call onClick in the actual Button component not pass it as a prop for better semantics ![screen](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/panel/passToButton.png)